primitives/blockchain/src/backend.rs

```diff
@@ -172,7 +172,7 @@ pub trait Backend<Block: BlockT>: HeaderBackend<Block> + HeaderMetadata<Block, E
 			if let Some(max_number) = maybe_max_number {
 				loop {
 					let current_header = self.header(BlockId::Hash(current_hash.clone()))?
-						.ok_or_else(|| Error::from(format!("failed to get header for hash {}", current_hash)))?;
+						.ok_or_else(|| Error::MissingHashInHeader(current_hash.to_string()))?;
```

Comment:
This is a really bad naming for the variant.

Reply:
Open for suggestions :>

Reply:
`MissingHeader`